js/components/report/score-box.js

```diff
@@ -5,19 +5,35 @@ export default class ScoreBox extends PureComponent {
     super(props);
   }
 
+  // The score property is a little confusing here
+  // when the ScoreBox is disabled changes to the score property will be shown
+  // when it is not disable, the score property just initializes the text field.
+  // If you need the score to update you'll need to recreate the ScoreBox or switch it
```

Comment:
If we wanted to update the `NumericTextField`'s  displayed value when its `intialValue` prop changed we could try using [`getDerivedStateFromProps(props, state)`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).

But I don't see a scenario where this approach is going to be a problem ATM.

Reply:
That is kind of how it was trying to work before, but that seems to be a React antipattern since it is mixing a controlled and uncontrolled component. More info about that is in this slack thread:
https://concord-consortium.slack.com/archives/C0303QUNU/p1574100016370300
and this React post:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html